vm-setup/roles/v1aX_integration_test/tasks/generate_templates.yml

```diff
@@ -0,0 +1,59 @@
+---
+  - name: Create a temporary manifest directory if it does not exist
+    file:
+      path: "{{ TEMP_GEN_DIR }}"
+      state: directory
+      mode: '0755'
+
+  - name: Generate clusterctl env file
+    template:
+      src: "{{ CRS_PATH }}/clusterctl_env_template_{{ IMAGE_OS|lower }}.rc"
+      dest: "{{ TEMP_GEN_DIR }}/clusterctl_env_{{ IMAGE_OS|lower }}.rc"
+
+  - name: Generate templates
+    shell: |
+      . {{ TEMP_GEN_DIR }}/clusterctl_env_{{ IMAGE_OS|lower }}.rc && {{ CLUSTERCTL_PATH}}/clusterctl config cluster {{ CLUSTER_NAME }} --kubernetes-version {{ KUBERNETES_VERSION }} > {{ TEMP_GEN_DIR }}/manifests.yaml
+
+  - name: Create a base dir
+    file:
+      path: "{{ TEMP_GEN_DIR }}/base"
+      state: directory
+      mode: '0755'
+
+  - name: Copy kustomization skeleton
+    copy:
+      src: "{{ CRS_PATH }}/kustomization.yaml"
+      dest: "{{ TEMP_GEN_DIR }}"
+
+  - name: Map replicas values
+    template:
+      src: "{{ CRS_PATH }}/replicas.yaml"
+      dest: "{{ TEMP_GEN_DIR }}/replicas.yaml"
```

Comment:
the number of replicase can be given as variables to clusterctl config. I am not sure what this whole kustomization part does.

Reply:
Fixed. Kustomization is need to split and merge manifests when separate provision steps are cluster, controlplane, worker